lib/random.php

```diff
@@ -296,14 +273,15 @@ function random_int($min, $max)
              * Apply mask
              */
             $val &= $mask;
+            $val += $valueShift;
 
             ++$attempts;
             /**
              * If $val is larger than the maximum acceptable number for
              * $min and $max, we discard and try again.
              */
-        } while ($val > $range);
-        return (int) ($min + $val);
+        } while (!is_int($val) || $val > $max || $val < $min);
```

Comment:
Do we care about side channels leaking `$min` and `$max` (but not the returned value) here?

`!is_int($val)` can only be true in the non-extended-range case when `$min` (`$valueShift`) is big and `$val & $mask` is enough to overflow the integer. Whether or not that branch is taken (and how often it is) leaks some information about `$min` and `$max`. But then again, so does the number of times this loop executes. So I'm guessing that's *not* something we're trying to protect against, and this is a non-issue.